### ISSUE TITLE
Missing Id in "trainval_im_names"

### ISSUE THREAD

**Issue by lennartmoritz** [stormyoung/PRAI-1581#2]
Thank you for fixing the previous inconsistencies.
But I believe that the last fix contains an error. I will elaborate the issue in the following.

**TL;DR: The list of images in "trainval_im_names" only contains 781 unique IDs but should contain 782**

Just for reference your paper states:

> Our PRAI-1581 dataset contains a total of 39,461 images for 1581 individual identities. We randomly divide the dataset into training and test set for only once and keep the partition ﬁxed in all the experiments for fair comparison. The ratio of training and testing set is roughly set to 1:1. The training set includes 19,523 images of 782 identities. And the rest of the dataset is used as the test set with 19,938 images of 799 identities. To be more speciﬁc, the numbers of query and gallery images are 4680 and 15,258 respectively.

### Problem:
Within the _partitions.pkl_ File there are multiple Objects that define the test/training split. In particular I extract the _pid2label_ dict and a list of _train_image_names_. The following is a simpliefied example of my code.
```
        with open(partition_path, "rb") as openfile:
            partition = pickle.load(openfile)
        train_image_names = partition["trainval_im_names"]
        pid2label = partition["trainval_ids2labels"]
```
When I relabel the training data and send it to my Re-ID framework (torchreid) for the actual training, the Programm crashes because there are only 781 instead of 782 unique IDs in the list of _train_image_names_.

Now the pid2label will label the last of the actually existing IDs from _train_image_names_ with the number 781 while the sum of actually existing labels is only 781. Therefore the new labeld index of 781 will be ouf of range since the Re-ID framework only expects the IDs in range [0, ... ,780].

### Which Id is missing:
I wrote a little script to find out which Person ID is actually missing. Its the Person with PID 382 / Label 195.
This is the output my Script will generate:

```
=> Loading train (source) dataset
Finished loading 39461 images.         Testimages: 19938 | Trainimages: 19523
There are 19938 elements in test_marks.
There are 15258 positive (1) entrys in test_marks.
There are 4680 negative (0) entrys in test_marks.
Found 0 duplicates in both sets
Missing 0 images from both sets
Only 781 PIDs in the training_set! But expected 782 PIDs.
The label 195 is missing.
The associated PID is 382.
-------------------------------------------------------------------
WARNING: PRAI1581 DATASET IS CORRUPTED! USING A WORKAROUND NOW!
 YOU WILL GET NON-REPRESENTATIVE DATA THIS WAY!
-------------------------------------------------------------------
=> Loaded PRAI1581
  ----------------------------------------
  subset   | # ids | # images | # cameras
  ----------------------------------------
  train    |   781 |    19523 |         2
  query    |   799 |     4680 |         2
  gallery  |   799 |    15258 |         2
  ----------------------------------------
```

I would appreciate it if you could update the data or let me know if I'm just using your Split incorrectly.

**Comment by stormyoung** [comment]
Hello, @lennartmoritz,
Thank you for your carefully check and good question. 
There are 781 IDs in train set. It is a mistake in our paper.
Thank you.
